### PR TITLE
Cleanup: Remove un-used product beta feature switch code

### PR DESCRIPTION
**Pull request by Ecarrion**
# Why

While working on the add-ons feature I noticed that we have some lingering code about the old product beta feature switch. This PR removes that stale code.

# Testing

Just make sure the CI passes

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerceTests/ViewRelated/Products/ProductDetailsFactoryTests.swift [hunk 1]
```diff
@@ -6,36 +6,14 @@ import Yosemite
 final class ProductDetailsFactoryTests: XCTestCase {
     // MARK: Simple product type
 
-    func test_factory_creates_product_form_for_simple_product_when_products_release_3_is_on() {
+    func test_factory_creates_product_form_for_simple_product() {
```
Comment (allendav) [inline WooCommerce/WooCommerceTests/ViewRelated/Products/ProductDetailsFactoryTests.swift:9]: The `expectation` below (`Wait for loading Products feature switch from app settings`) should be updated as well (1 of 2)

File: WooCommerce/WooCommerceTests/ViewRelated/Products/ProductDetailsFactoryTests.swift [hunk 2]
```diff
@@ -46,35 +24,14 @@ final class ProductDetailsFactoryTests: XCTestCase {
 
     // MARK: External/affiliate product type
 
-    func test_factory_creates_product_form_for_affiliate_product_when_products_release_3_is_on() {
+    func test_factory_creates_product_form_for_affiliate_product() {
```
Comment (allendav) [inline WooCommerce/WooCommerceTests/ViewRelated/Products/ProductDetailsFactoryTests.swift:27]: The `expectation` below (`Wait for loading Products feature switch from app settings`) should be updated as well (2 of 2)

File: WooCommerce/WooCommerceTests/ViewRelated/Products/ProductDetailsFactoryTests.swift [hunk 3]
```diff
@@ -85,34 +42,14 @@ final class ProductDetailsFactoryTests: XCTestCase {
...
-                                             forceReadOnly: false) { viewController in
-                                                // Assert
-                                                XCTAssertTrue(viewController is ProductFormViewController<ProductFormViewModel>)
-                                                expectation.fulfill()
-        }
-        waitForExpectations(timeout: Constants.expectationTimeout, handler: nil)
-    }
-
-    func test_factory_creates_product_form_for_grouped_product_when_products_release_3_is_off() {
+    func test_factory_creates_product_form_for_grouped_product() {
```
Comment (allendav) [inline WooCommerce/WooCommerceTests/ViewRelated/Products/ProductDetailsFactoryTests.swift:45]: The `expectation` below (`Wait for loading Products feature switch from app settings`) should be updated as well (and thereafter, multiple instances)

**Review by allendav — APPROVED** [review]
A few expectation descriptions should be updated, but otherwise :shipit:

**Comment by Ecarrion** [comment]
@allendav Thanks for spotting those, I have updated them!

**Comment by allendav** [comment]
Oh, and I think you can remove these too, right?

https://github.com/woocommerce/woocommerce-ios/pull/4490/files#diff-589afa6af8ad1530b89e871a8a3387339026dd4e6b57d837ae1613fad6084fd5R539-R541

Sorry I forgot to mention those the first time!

**Pull request merged**